src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<K extends Machine> {
```

Comment:
Generic <K> usually used in Map, where K is key and V is value.
Generic <T> , what means Type suits here better

Reply:
done